[Q]: Step17-1

park78951 opened the pull request:
step17-1 미션 리뷰 부탁드립니다.

Babel이 하는 일을 정확하게 이해하고자 노력했지만 마지막에 polyfill과 core가 함께 쓰이지 않는다고 말씀해주셔서 찾아보니 @babel/polyfill이 필요 없었습니다.. 다시 볼때마다 새로워서 앞으로도 계속 공부해야할 듯싶네요...

[데모링크](https://park78951.github.io/codesquad-Q/)

park78951 committed:
Add components, styles and fetcing datas

- add components depending on each component's feature.
- apply styled components
- add code to fetch datas from a provided api for this mission
- revise babel.config.js and add plugin runtime and transform runtime to
async / await

park78951 committed:
Add feat handling errors

- add catch at the part fetching datas by axios
- add gitignore
- polish codes

park78951 committed:
Refactor and polish codes:

- add cursor on button by styled components
- remove comments on webpack

park78951 committed:
Rename InputBar and polish codes

- Rename InputBar.js to Input.js
- Remove unecessary props on Input and TodoList

park78951 committed:
Polish Codes

- Add missed props

park78951 committed:
Handle errors by using try catch

- Change catch from then chains to try-catch to handle errors.
- Add error.js for error messages to use when error is catched.

park78951 committed:
Build for deploy

- Add on scripts of webpack.config: build, predeploy, deploy
- Create build files by webpack.

park78951 committed:
Revise main in package.json for deploy

park78951 committed:
Remove files created for deploying test

- Add build and deploy commands for deploy.
- Remove unnecessary codes.

park78951 committed:
Organize babels and dependencies.

- remove unnecessary things in  dependencies and scripts

crongro commented:
polyfill 전략이 좀 어려워요. babel을 쓰지 않으면 사실 필요한 polyfill을 내가 만들거나 또 다른 라이브러리를 가져와서 추가(import)해주는 쉬운방법도 있죠.
하지만 일일이 추가해주는 게 번거로운 일이라, 종합선물세트 babel의 core-js가 좋아보이는거 같네요.

crongro commented on package.json:
```diff
@@ -0,0 +1,33 @@
+  "license": "MIT",
+  "dependencies": {
+    "@babel/runtime": "^7.5.5",
+    "axios": "^0.19.0",
```
axios 를 fetch API와 비교해보시면서 장/단점을 찾아보면 좋겠네요.

crongro commented on src/components/App.js:
```diff
@@ -0,0 +1,45 @@
+import React, {Component} from 'react';
+import Input from './Input/Input';
+import TodoList from './TodoList/TodoList';
+import ShowContentsBtn from './ShowButton/ShowContentsBtn';
```
클래스 컴포넌트를 대문자로 시작하니  함수같아요.
모든 컴포넌트는 명사로..

crongro commented on src/components/App.js:
```diff
@@ -0,0 +1,45 @@
+  }
+
+  componentDidMount() {
+    this.fetchMyTodoList();
```
componentDidMount 에서는 또 다른 task를 할수도 있긴하죠.
그렇기 때문에 여러가지 다른 로직이 뒤섞일수도 있을거에요.
fetchMyTodoList처럼 메서드를 분리하면 그럴일이 없어지겠죠. 잘 분리했어요~

crongro commented on src/components/App.js:
```diff
@@ -0,0 +1,45 @@
+
+  render() {
+    return(
+      <Div>
```
항상 root node를 만들어야해서 div로 감싸는 경우가 많은데요.
진짜 div가 필요한 것이 아니면, <> </>로 감싸주는 문법을 사용할 수도 있다고 하네요.

crongro commented on src/components/TodoList/Contents.js:
```diff
@@ -0,0 +1,50 @@
+  padding: 0;
+`
+
+const P = styled.p`
```
이렇게 각각 다 styled을 부여하면 상당히 힘들고, 계속 사용자정의 컴포넌트를 만들어야할텐데요.
여길참고해서 구현해보세요.

https://codesandbox.io/embed/bitter-lake-o65bo

park78951 replied:
오..... 자세한 설명 감사합니다 ㅜㅜㅜ 이렇게 수정하는 방법은 생각 못했었네요. 참고해서 수정했습니다!

crongro commented on webpack.config.js:
```diff
@@ -0,0 +1,25 @@
+      }
+    ]
+  },
+  plugins: [
```
간단히 잘 설정했고요.
웹팩은 module과 plugins의 차이를 설명해보세요! 에 대답하실수 있으면 좋겠습니다

crongro reviewed (changes requested):
프론트엔드 개발 설정이 어렵죠.. babel/webpack공식가이드를 보면서 공부하면 좋지만 며칠 걸리거에요.
우리가 배운 내용위주로 우선이해하시고 확장해나가면 좋죠.

수정요청합니다.
styled-component를 링크 추가한 거 참고해서 수정해보세요!

park78951 committed:
Modify the way of styled components

- use element tag name for the children elements
- change showContentsBtn -> ContentsBtn & Input -> InputBar

park78951 committed:
Revise errors:

- styled components with wrong tags
- the wrong route of inputBar

park78951 commented:
말씀해주신 대로 수정했습니다. styled components의 사용이 굉장히 자유롭네요.. 더욱 공부해야 할 것 같습니다. axios와 fetch의 차이점, webpack의 module과 plugins의 차이점에 대해서 따로 정리해보도록 하겠습니다.

crongro reviewed (approved):
네 styled-component도 꽤 막강하죠.
props속성을 전달받아서 동적으로  css value값을 결정하는, 기능도 있으니 참고하세요.
